### ISSUE TITLE
Assert that a toast appeared

### ISSUE THREAD

**Issue by estefafdez** [SchibstedSpain/Barista#119]
I would like to have an assert method to check if the toast message is displayed properly.

**Comment by rocboronat** [comment]
Thanks for the issue! In the meantime, you can use native Espresso: https://stackoverflow.com/questions/28390574/checking-toast-message-in-android-espresso

**Comment by estefafdez** [comment]
Great, thank you!

**Comment by Sloy** [comment]
We've been discussing this internally, and I'm sorry to say that we think it's best not to include this feature in Barista.

TL;DR: The reason is that there's no deterministic way of testing it. Although the proposed solution might work for many cases, it can introduce timing issues and make your tests more flaky. 

[Here's the response](https://groups.google.com/forum/#!activity/android-test-kit-discuss/cuQASd3PdpgJ) to a similar question from one of Espresso's engineers: 
Basically the proposed solution is to use a wrapper around Toast, so you can use Mockito to intercept and verify the invocation to your wrapper.
depending on the project it would be done differently,
This would be done differently depending on the project and requires client-side code changes, so we think it's best to leave it outside Barista.

I'm sorry we can't help you with this @estefafdez, but at least I hope the answer was helpful.
